node_modules/oae-principals/tests/test-guests.js

```diff
@@ -0,0 +1,437 @@
+
+            // Create another content item and invite the guest again
+            RestAPI.Content.createLink(simong.restContext, 'Apereo', 'Apereo', 'public', 'http://www.apereo.org', [], [], [], function(err, link2) {
+                assert.ok(!err);
+                RestAPI.Content.shareContent(simong.restContext, link2.id, [email], function(err) {
+                    assert.ok(!err);
+
+                    // Create a guest account with the token that was emailed to the guest
+                    PrincipalsTestUtil.signIntoGuestAccountWithToken(token, function(loggedinGuest) {
+
+                        // Verify both the content items are in the guest's content library
+                        RestAPI.Content.getLibrary(loggedinGuest.restContext, loggedinGuest.user.id, null, null, function(err, data) {
+                            assert.ok(!err);
+                            assert.strictEqual(data.results.length, 2);
+                            assert.strictEqual(data.results[0].id, link2.id);
+                            assert.strictEqual(data.results[1].id, link1.id);
+                            return callback();
+                        });
+                    });
+                });
+            });
+        });
+    });
+
+    /**
+     * Test that verifies that guests are created on the guests tenant if no tenant email domain matches
+     */
+    it('verify that guests are created on the guests tenant if no tenant email domain matches', function(callback) {
+        TestsUtil.generateTestUsers(camAdminRestContext, 1, function(err, users, simong) {
+            assert.ok(!err);
+
+            RestAPI.Content.createLink(simong.restContext, 'Apereo', 'Apereo', 'public', 'http://www.apereo.org', [], [], [], function(err, link) {
+                assert.ok(!err);
+
+                // Share the link with a guest user
+                var email = TestsUtil.generateTestEmailAddress();
+                RestAPI.Content.shareContent(simong.restContext, link.id, [email], function(err) {
+                    assert.ok(!err);
+
+                    // Verify that the user account was provisioned on the correct tenant
+                    RestAPI.Content.getMembers(simong.restContext, link.id, null, null, function(err, members) {
+                        assert.ok(!err);
+                        var viewer = _.findWhere(members.results, {'role': 'viewer'});
+                        assert.ok(viewer);
+                        assert.strictEqual(viewer.profile.tenant.alias, 'guests');
+                        return callback();
+                    });
+                });
+            });
+        });
+    });
+
+    /**
+     * Test that verifies that guests are created on the correct tenant if a tenant email domain matches
+     */
+    it('verify that guests are created on the correct tenant if a tenant email domain matches', function(callback) {
```

Comment:
Should verify this both with an existing and non-existing one. We should probably also verify with an existing one when the email domain actually matches a different tenant.